### DISCUSSION TITLE
Getting Server Error on Netbox dashboard

### DISCUSSION THREAD

**Discussion by vinoymohan** [netbox-community/netbox#13246] [Q&A]
Hello Everyone,

I am getting the below error when i try to add an external Postgres DB to the netbox. Until i attach an external DB i was able to see the netbox dashboard and able to login with my LDAP user credentails. Kindly help on this

Server Error
A database programming error was detected while processing this request. Common causes include the following:

Database migrations missing - When upgrading to a new NetBox release, the upgrade script must be run to apply any new database migrations. You can run migrations manually by executing python3 manage.py migrate from the command line.

Unsupported PostgreSQL version - Ensure that PostgreSQL version 10 or later is in use. You can check this by connecting to the database using NetBox's credentials and issuing a query for SELECT VERSION().

The complete exception is provided below:

<class 'django.db.utils.ProgrammingError'>

relation "extras_dashboard" does not exist
LINE 1: ...board"."layout", "extras_dashboard"."config" FROM "extras_da...
                                                             ^

Python version: 3.8.10
NetBox version: 3.5.6
If further assistance is required, please post to the [NetBox discussion forum](https://github.com/netbox-community/netbox/discussions) on GitHub

**Comment by J-DubApps** [comment]
My exact scenario & situation. Did you figure it out?  I suspect it occurs on systems where you perform a Netbox upgrade, but did not first perform apt-get upgrade runs prior - and so Postgres DB didn't get some new fancy added field or relation.  Upgrading Postgres & re-running the upgrade.sh won't help, either.  Nor will a manage.py migrate run.  Just dead in the water.   

extras_dashboard relation missing = your entire install is ruined.  Or, that is where I am at the moment - anyway...  I will experiment and return with an update if I solve it.  Sorry you did not get a reply here.

**Reply by J-DubApps** [reply]
My latest: I do see the legacy ASN data message go by, where upgrade.sh appears to stop on db upgrade steps.  So I am going to run "export NETBOX_DELETE_LEGACY_DATA=1" on the theory that db upgrades did not proceed due to it finding legacy ASN data.

**Reply by J-DubApps** [reply]
Yup, that was my issue -- whereas I thought it was my psql version originally, it was that I had two providers in our db with legacy ASN info.  Now I have DB objects created, related to the upgrade.  I don't know if this will fix things for your scenario, or if you want to risk deleting this field (I only had two records using it, didn't care, but you may have many more...).   
The key is watching all output of upgrade.sh, in my case I figured out that it never got to the db upgrade steps due to my legacy ASN data.  Once I gave upgrade.sh permission to delete that legacy data, I got the DB upgrade routine to run by re-runing upgrade.sh.  Cmd was: 

sudo NETBOX_DELETE_LEGACY_DATA=1 ./upgrade.sh

So I am fixed.  Use this post with caution, I make no promises but it got my upgrade completed!
